tests/qnodes/test_qnode_base.py

```diff
@@ -986,6 +1005,50 @@ def test_decompose_queue(self, operable_mock_device_2_wires):
         assert res[4].name == "RX"
         assert res[4].parameters == [6]
 
+    def test_decompose_queue_inv(self, operable_mock_device_2_wires_with_inverses):
+        """Test that decompose queue works correctly
+        when an operation exists that can be decomposed"""
+
+        queue = [
+            qml.Rot(0, 1, 2, wires=0).inv(),
+            qml.U3(3, 4, 5, wires=0).inv(),
+            qml.RX(6, wires=0).inv(),
+        ]
+
+        res = decompose_queue(queue, operable_mock_device_2_wires_with_inverses)
+
+        print([op.name for op in res])
+        print([op.parameters for op in res])
+
+        assert len(res) == 9
+
+        assert res[0].name == "RZ.inv"
+        assert res[0].parameters == [2]
+
+        assert res[1].name == "RY.inv"
+        assert res[1].parameters == [1]
+
+        assert res[2].name == "RZ.inv"
+        assert res[2].parameters == [0]
+
+        assert res[3].name == "PhaseShift.inv"
+        assert res[3].parameters == [4]
+
+        assert res[4].name == "PhaseShift.inv"
+        assert res[4].parameters == [5]
+
+        assert res[5].name == "RZ.inv"
+        assert res[5].parameters == [-5]
+
+        assert res[6].name == "RY.inv"
+        assert res[6].parameters == [3]
+
+        assert res[7].name == "RZ.inv"
+        assert res[7].parameters == [5]
+
+        assert res[8].name == "RX.inv"
+        assert res[8].parameters == [6]
+
```

Comment:
It might be worth including a `test_decompose_queue_inv_recursive` test as well, where there are certain operations in the decomposition, which themselves need decomposing as well. I think one of the most steps of recursion would be having such a test using `CRY` with the line of decompositions on one branch being: `U3` -> `Rot`  -> basic rotations.

Reply:
The current test here has recursion already, so I guess you're referring to adding a test with recursion that has no inverses?

Reply:
I implemented such a test now!

Reply:
By the way, why is the decomposition of `CRY` going via `U3`? If `U3` and `Rot` are not supported, this adds a lot of overhead. I think directly putting
```
CRY -> RY CNOT RY
```
makes more sense.

Reply:
Oh, in that case, that was my bad, sorry about that! Great thanks so much! :)